plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -327,6 +330,21 @@ function wc_body_class( $classes ) {
 		$classes[] = 'woocommerce-account';
 		$classes[] = 'woocommerce-page';
 
+		$account_page_id = get_option( 'woocommerce_myaccount_page_id' );
+		$account_page_slug = false;
+
+		if ( !empty( $account_page_id ) ) {
+
+			$account_page_slug = get_post_field( 'post_name', $account_page_id );
+
+		}
+
+		if ( basename( $wp->request ) == $account_page_slug && is_user_logged_in() ) {
```

Comment:
This `if` could be nested into the first one, right? Then the initial `$account_page_slug = false;` could be removed.

Reply:
@Konamiman I have refactored this as per the feedback, please can you recheck?